app/controllers/userdata_controller.rb

```diff
@@ -0,0 +1,113 @@
+class UserdataController < ApplicationController
```

Comment:
So the plan is to wait until we merge unattended controller changes and then to move this logic into it?

Reply:
I don't think we should actually merge the controllers, but this controller shares a lot of similar code with unattended controller. We have extracted most of the duplicate code out of the controller, so we should be able to reuse that fairly easily.
I know, you don't like the separate controller, but I think unattended controller is already way to complex. I prefer a nice and small controller for the user data stuff.

Reply:
Can you give me more arguments? Because the way I see this: user-data is just a template, very similar situation like with iPXE template you recently added. Why to bother with two controllers?

Reply:
Another thing is that the cloud-init templates need to be accessible even when the host is not in build mode iirc. That will even be another exception that we have to patch into the controller.

Reply:
Well, my argument is security. Unattended controller is special one, it skips all the authorization, authentication, CSRF and other protections. It is essentially open window over HTTP into Foreman. Now, I would love to see just one such a window instead of two in the project. Technically this is a template rendering, except the context is a bit customized.

I don't see this as a blocker tho, if you feel like new controller is the right way then please remove duplicated code as much as possible. Two major ones I see are skipping various filters and actions and the host find mechanism (this has been already refactored in the other PR).